lib/faraday/adapter/net_http_persistent.rb

```diff
@@ -17,8 +17,11 @@ def with_net_http_connection(env)
             define_method(:password) { proxy[:password] }
           end if proxy[:user]
         end
-
-        yield Net::HTTP::Persistent.new 'Faraday', proxy_uri
+        if Net::HTTP::Persistent::VERSION >= '3'
```

Comment:
Instead of checking a version string like this, could we simply check the arity of the constructor?

```
Net::HTTP::Persistent.method(:new).arity
```

We could detect how many arguments does the constructor expect this way, and switch between two different invocation styles without being dependent on the version.